tests/simulator/agenda_based/test_interaction_model.py

```diff
@@ -54,3 +77,52 @@ def test_get_next_dialogue_acts(im_crsv1: InteractionModel) -> None:
         im_crsv1.INTENT_DISCLOSE,
         [SlotValueAnnotation("DIRECTOR", "Steven Spielberg")],
     )
+
+
+def test_intent_types(im_crsv1: InteractionModel) -> None:
+    """Tests methods checking the type of intents."""
+    assert im_crsv1.is_agent_intent_elicit(Intent("ELICIT")) is False
+    assert im_crsv1.is_agent_intent_inquire(Intent("INQUIRE")) is True
+    assert im_crsv1.is_agent_intent_set_retrieval(Intent("REVEAL")) is True
+
+
+def test_is_transition_allowed(monkeypatch, im_crsv1: InteractionModel) -> None:
+    monkeypatch.setattr(
+        im_crsv1, "_current_dialogue_acts", [DialogueAct(Intent("DISCLOSE"))]
+    )
+    dialogue_acts_allowed = [
+        DialogueAct(Intent("INQUIRE")),
+        DialogueAct(Intent("REVEAL")),
+    ]
+    dialogue_acts_not_allowed = [DialogueAct(Intent("END"))]
+
+    assert im_crsv1._is_transition_allowed(dialogue_acts_allowed) is True
+    assert im_crsv1._is_transition_allowed(dialogue_acts_not_allowed) is False
+
+
+def test_sample_next_user_dialogue_acts(
+    caplog, im_crsv1: InteractionModel, information_need: InformationNeed
+) -> None:
+    agent_dialogue_acts = [
+        DialogueAct(Intent("GREETING")),
+        DialogueAct(Intent("ELICIT")),
+    ]
+    user_dialogue_acts = im_crsv1._sample_next_user_dialogue_acts(
+        information_need, agent_dialogue_acts
+    )
+    assert len(user_dialogue_acts) == 1
+    assert (
+        "Transition matrix does not contain agent intent: GREETING"
+        in caplog.text
+    )
+
+    agent_dialogue_acts = [
+        DialogueAct(Intent("ELICIT")),
+    ]
+    user_dialogue_acts = im_crsv1._sample_next_user_dialogue_acts(
+        information_need, agent_dialogue_acts
+    )
+    assert user_dialogue_acts == [
```

Comment:
Perhaps just leave a note here saying that the sampling is non-deterministic, and thus different runs of the test may yield different results.

Reply:
Done.